ios/PaymentSheetAppearance.swift

```diff
@@ -0,0 +1,163 @@
+        colors.componentText = StripeSdk.buildUserInterfaceStyleAwareColor(key: "componentText", lightParams: lightModeParams, darkParams: darkModeParams) ?? PaymentSheet.Appearance.default.colors.componentText
+        colors.componentPlaceholderText = StripeSdk.buildUserInterfaceStyleAwareColor(key: "placeholderText", lightParams: lightModeParams, darkParams: darkModeParams) ?? PaymentSheet.Appearance.default.colors.componentPlaceholderText
+        colors.icon = StripeSdk.buildUserInterfaceStyleAwareColor(key: "icon", lightParams: lightModeParams, darkParams: darkModeParams) ?? PaymentSheet.Appearance.default.colors.icon
+        colors.danger = StripeSdk.buildUserInterfaceStyleAwareColor(key: "error", lightParams: lightModeParams, darkParams: darkModeParams) ?? PaymentSheet.Appearance.default.colors.danger
+        
+        return colors
+    }
+    
+    private func buildShadow(params: NSDictionary) -> PaymentSheet.Appearance.Shadow {
+        var shadow = PaymentSheet.Appearance.Shadow()
+        
+        if let color = StripeSdk.buildUserInterfaceStyleAwareColor(key: "color", lightParams: params, darkParams: params) {
+            shadow.color = color
+        }
+        if let opacity = params["opacity"] as? CGFloat {
+            shadow.opacity = opacity
+        }
+        if let radius = params["borderRadius"] as? CGFloat {
+            shadow.radius = radius
+        }
+        if let offsetParams = params["offset"] as? NSDictionary {
+            if let x = offsetParams["x"] as? CGFloat, let y = offsetParams["y"] as? CGFloat {
+                shadow.offset = CGSize(width: x, height:y)
+            }
+        }
+
+        return shadow
+    }
+    
+    private func buildPrimaryButton(params: NSDictionary) throws -> PaymentSheet.Appearance.PrimaryButton {
+        var primaryButton = PaymentSheet.Appearance.PrimaryButton()
+        
+        if let fontName = (params["font"] as? NSDictionary)?["family"] as? String {
+            guard let customFont = UIFont(name: fontName, size: UIFont.systemFontSize) else {
+                throw PaymentSheetAppearanceError.missingFont(fontName)
+            }
+            primaryButton.font = customFont
+        }
+        if let shapeParams = params["shapes"] as? NSDictionary {
+            if let borderRadius = shapeParams["borderRadius"] as? CGFloat {
+                primaryButton.cornerRadius = borderRadius
+            }
+            if let borderWidth = shapeParams["borderWidth"] as? CGFloat {
+                primaryButton.borderWidth = borderWidth
+            }
+            if let shadowParams = shapeParams["shadow"] as? NSDictionary {
+                primaryButton.shadow = buildShadow(params: shadowParams)
+            }
+        }
+        if let colorParams = params["colors"] as? NSDictionary {
+            if (colorParams.object(forKey: "light") != nil && colorParams.object(forKey: "dark") == nil ||
```

Comment:
Same comment here for simplifying the logic if possible

Reply:
same goes for here as [above](https://github.com/stripe/stripe-react-native/pull/940#discussion_r878565979)